src/Data/List/Relation/Binary/Pointwise.agda

```diff
@@ -166,6 +166,15 @@ tabulate⁻ {n = suc n} (x∼y ∷ xs∼ys) (fsuc i) = tabulate⁻ xs∼ys i
 ++-cancelʳ {xs = xs}     (y ∷ ys) []       eq   =
   contradiction (≡.trans (≡.sym (length-++ (y ∷ ys))) (Pointwise-length eq)) (m≢1+n+m (length xs) ∘ ≡.sym)
 
+module _ (rfl : Reflexive R) where
+
+  ++⁺ʳ : ∀ xs → Pointwise R ys zs → Pointwise R (xs ++ ys) (xs ++ zs)
```

Comment:
Aren't you inconsistent between these two wrt to where you put the quantified variable?

Also, I would think that the `xs` (here) and `zs` (below) could both be implicit? They're not needed in the proofs at all, which is a strong sign that they are not strongly relevant.

Reply:
I'll leave this TODO for now, but I suspect that something similar is at play in the problems with #2415 ...? (Forward proof permits weaker quantification, while backward proof demands stronger thanks to the semantics of matching/unification?)

Reply:
Have now added revised types for the new lemmas, and note the following:
* `++⁺ʳ` now has a type corresponding to the explicitly quantified version of `Algebra.Definitions.LeftCongruent`
* `++⁺ˡ` now has a type corresponding to the explicitly quantified version of `Algebra.Definitions.RightCongruent`

Which nomenclature/notation should be taken as canon?